client/utils/utils.go

```diff
@@ -49,12 +58,62 @@ func SendTx(txCtx authctx.TxContext, cliCtx context.CLIContext, msgs []sdk.Msg)
 		return err
 	}
 
+	txCtx, err = enrichCtxWithGasIfGasAuto(txCtx, cliCtx, cliCtx.FromAddressName, passphrase, msgs)
+	if err != nil {
+		return err
+	}
+
 	// build and sign the transaction
 	txBytes, err := txCtx.BuildAndSign(cliCtx.FromAddressName, passphrase, msgs)
 	if err != nil {
 		return err
 	}
-
 	// broadcast to a Tendermint node
 	return cliCtx.EnsureBroadcastTx(txBytes)
 }
+
+func enrichCtxWithGasIfGasAuto(txCtx authctx.TxContext, cliCtx context.CLIContext, name, passphrase string, msgs []sdk.Msg) (authctx.TxContext, error) {
```

Comment:
Why don't we just do this check in the exported function. The flow of unexported function wrapping the exported function seems odd to me, when this is functionality we probably want in the external API, given that its what we have in the CLI.

Reply:
I wrapped the exported function to reduce cyclomatic complexity and make linter happy